apps/web/app/(booking-page-wrapper)/team/[slug]/[type]/page.tsx

```diff
@@ -13,7 +14,20 @@ import { getServerSideProps } from "@lib/team/[slug]/[type]/getServerSideProps";
 import LegacyPage from "~/team/type-view";
 import type { PageProps as LegacyPageProps } from "~/team/type-view";
 
+import CachedTeamBooker, { generateMetadata as generateCachedMetadata } from "./pageWithCachedData";
+
+async function isCachedTeamBookingEnabled(searchParams: SearchParams): Promise<boolean> {
+  const featuresRepository = new FeaturesRepository();
+  const isGloballyEnabled = await featuresRepository.checkIfFeatureIsEnabledGlobally(
+    "team-booking-page-cache"
+  );
+  return isGloballyEnabled && searchParams.experimentalTeamBookingPageCache === "true";
```

Comment:
@keithwillcode I took this approach of global feature flag + search param instead of per-team feature flag. The reason is because team id (needed for checking per-team feature flag) isn't immediately available in RSC and an extra fetch is suboptimal